File: src/main/java/uk/co/alt236/apkcompare/repo/dex/model/Parameter.java
package uk.co.alt236.apkcompare.repo.dex.model;

import org.jf.dexlib2.iface.MethodParameter;

public class Parameter {

    private final DexClassType type;
    private final String name;

    Parameter(MethodParameter param) {
        name = param.getName();
        type = new DexClassType(param.getType());
    }

    public DexClassType getType() {
        return type;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "Parameter{" +
                "type='" + type + '\'' +
                ", name='" + name + '\'' +
                '}';
    }
}

File: apk/src/main/java/uk/co/alt236/apk/repo/dex/model/Parameter.kt
package uk.co.alt236.apk.repo.dex.model

import org.jf.dexlib2.iface.MethodParameter

data class Parameter internal constructor(val name: String?, val type: DexClassType) {
    internal constructor(param: MethodParameter) : this(param.name, DexClassType(param.type))
}
